hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/signature/AWSSignatureProcessor.java

```diff
@@ -67,7 +78,13 @@ public SignatureInfo parseSignature() throws OS3Exception {
 
     SignatureInfo signatureInfo = null;
     for (SignatureParser parser : signatureParsers) {
-      signatureInfo = parser.parseSignature();
+      try {
+        signatureInfo = parser.parseSignature();
+      } catch (MalformedResourceException e) {
+        AuditMessage message = buildAuthFailureMessage(e);
+        AUDIT.logAuthFailure(message);
+        throw S3ErrorTable.newError(MALFORMED_HEADER, e.getResource());
+      }
       if (signatureInfo != null) {
         break;
       }
```

Comment:
I see this exception in the [log](https://github.com/apache/ozone/suites/7311716169/artifacts/295519364):

```
s3g_1        | Jul 11, 2022 10:46:07 PM org.glassfish.jersey.internal.Errors logErrors
s3g_1        | WARNING: The following warnings have been detected: WARNING: Unknown HK2 failure detected:
s3g_1        | MultiException stack 1 of 1
s3g_1        | javax.ws.rs.WebApplicationException: The authorization header you provided is invalid.
s3g_1        | 	at org.apache.hadoop.ozone.s3.OzoneClientProducer.wrapOS3Exception(OzoneClientProducer.java:141)
s3g_1        | 	at org.apache.hadoop.ozone.s3.OzoneClientProducer.getSignature(OzoneClientProducer.java:102)
...
s3g_1        | Caused by: org.apache.hadoop.ozone.s3.exception.OS3Exception
s3g_1        | 	at org.apache.hadoop.ozone.s3.exception.S3ErrorTable.newError(S3ErrorTable.java:139)
s3g_1        | 	at org.apache.hadoop.ozone.s3.exception.S3ErrorTable.newError(S3ErrorTable.java:126)
s3g_1        | 	at org.apache.hadoop.ozone.s3.signature.AWSSignatureProcessor.parseSignature(AWSSignatureProcessor.java:86)
```

instead of the `AUTH` failure in audit.

Triggered by:

https://github.com/apache/ozone/blob/843fac2fb646eecfc33103fdb16eaf77a66ca062/hadoop-ozone/dist/src/main/smoketest/s3/bucketlist.robot#L43-L46

Reply:
Thanks for looking at the logs. Actually, the audit log was there but the action is not properly named. I've just submit an update to fixed this. 

The warning logs is another problem. It's from the fact that S3G parses and links S3 signature to Endpoints using dependency injection. i.e.

```
@RequestScoped
public class OzoneClientProducer {
  @Produces
  public S3Auth getSignature() {} 
}
```

```
public abstract class EndpointBase implements Auditor {

  @Inject
  private OzoneClient client;
```

The resolution of `S3Auth` is done by HK2 (Jersey lightweight IoC framework). When HK2 tries to create an `S3Auth` instance and gets exceptions for invalid/malformed signatures, it always logs the exceptions as warnings before passing to Jersey to resolve the exceptions to 4xx responses. I found no way to override this behavior. 

To avoid the warning logs for those user event exceptions, we can either refactor to avoid using DI for creating and injecting `S3Auth`, or just disable the those warning logs using log config (note that Jersey just doesn't use log4j but Java Util Logging, so this may be some complication). Think I should log this in another JIRA.

Reply:
Thanks @duongnguyen0 for digging into the exception problem.

> I should log this in another JIRA.

:+1: